api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreServiceImpl.java

```diff
@@ -171,14 +154,17 @@ public Patient registerPatient(Patient patient, List<Relationship> relationships
 				} else {
 					throw new APIException("Only one side of a relationship should be specified");
 				}
-				
+
 				personService.saveRelationship(relationship);
 			}
 		}
-		
+
 		EventMessage eventMessage = new EventMessage();
 		eventMessage.put(RegistrationCoreConstants.KEY_PATIENT_UUID, patient.getUuid());
+
+		//TODO incorrect set of "UUID" each time it generates new RANDOM value, instead of set uuid of creator.
```

Comment:
@djazayeri I'm not sure that I know how to fix id, it seems for me that "creator" is setting in incorrect way to patient, that's why he has incorrect "UUID". 
It seems like someone have created "Creator" user, instead of get it from DB. So, that "someone" set properly the ID field, but haven't set properly UUID field.
Actually, if someone is using this value around - it use INCORRECT value. So, we have to find the way how to correctly set UUID, instead of just remove this logic. :)